web_src/js/features/common-global.js

```diff
@@ -60,6 +60,57 @@ export function initGlobalButtonClickOnEnter() {
   });
 }
 
+async function fetchFormAction(e) {
+  if (!e.target.classList.contains('form-fetch-action')) return;
+
+  e.preventDefault();
+  const formEl = e.target;
+  if (formEl.classList.contains('loading')) return;
+
+  // TODO: fine tune the UI feedback
+  formEl.classList.add('loading');
+  e.submitter?.classList.add('loading');
+
+  const formMethod = formEl.getAttribute('method') || 'get';
+  const formActionUrl = formEl.getAttribute('action');
+  const formData = new FormData(formEl);
+  const [submitterName, submitterValue] = [e.submitter?.getAttribute('name'), e.submitter?.getAttribute('value')];
+  if (submitterName) {
+    formData.append(submitterName, submitterValue);
+  }
+
+  const req = {method: formMethod.toUpperCase(), headers: {'X-Csrf-Token': csrfToken}};
+  if (formMethod === 'GET') {
```

Comment:
```suggestion
  if (formMethod === 'get') {
```

form `method` in HTML is actually specced lowercase and treated case-insentively, but we have a number violations in HTML. I will eanble the linter rule for it in https://github.com/go-gitea/gitea/pull/25212 and fix the issues.

Reply:
https://github.com/go-gitea/gitea/pull/25212 now lowercases all form methods, but still good to keep it case-insensitive here regardless.

Reply:
Case-insensitive is a must. ps: this PR is incomplete, I haven't finished it nor really tested it, the code/logic will change a lot IMO.